tests/test_data_types.py

```diff
@@ -64,6 +59,58 @@ def test_bind_image():
         with pytest.raises(RuntimeError):
             wb_image.bind_to_run(run, 'stuff', 10)
 
+full_box = {
+    "position": {
+        "middle" : (100,100), "width" : 40, "height": 20 
+        },
+    "class_label": "car",
+    "box_caption": "This is a big car",
+    "scores": {
+    "acc": 0.3
+    }
+}
+
+
+optional_keys = ["class_label", "box_caption", "scores"]
+boxes_with_removed_optional_args = [dissoc(full_box, k) for k in optional_keys]
+
+def test_bounding_box_to_json():
+    box = BoundingBoxes2D(data)
+    json = box.to_json()
+    assert json["_type"] == "boxes2D"
+    # TODO: Test filename
+
+def test_boxes_to_json():
+    boxes = [BoundingBoxes2D(b) for b in boxes_with_removed_optional_args]
+    json = boxes.to_json()
+    for box in boxes: 
+        assert box["_type"] == "boxes2D"
+
+def test_image_accepts_boxes():
+    img = wandb.Image(image, boxes=[full_box])
+    # TODO: Test that file is dumped
+
+standard_mask = {
+    "mask_data": np.array([[1,2,2,2], [2,3,3,4], [4,4,4,4], [4,4,4,2]]),
+    "class_labels": { 
+        1: "car",
+        2: "pedestrian", 
+        3: "tractor", 
+        4: "cthululu" 
+    }
+}
+
+
+def test_masks_to_json():
+    mask = ImageMask(standard_mask)
+    json = mask.to_json()
+    assert json["_type"] == "boxes2D"
+    assert len(json["values"]) ==64
+
+def test_image_accepts_masks():
+    img = wandb.Image(image, masks=[standard_mask])
+    # TODO: Fill out test
+    assert img != False
```

Comment:
Would be good to assert the resulting Image json has appropriate filename pointers to the new mask pngs.

Reply:
@nbardy we should fill this out before releasing

Reply:
Just pushed new tests